### PR TITLE
CS Journeys upsell

### PR DESCRIPTION
**Pull request by cforkish**
the primary purpose of this PR is to add a teacher resource card to upsell CS Journeys. the new card includes a "_New!_" callout next to the title, so that functionality needed to be added to `ImageResourceCard` and `ResourceCard`.

when adding the new image for the CS Journeys card, i discovered that the implementation relied on using mostly-transparent images that were the full size of the card, along with floats and absolute positioning. i refactored the component to work with "actual size" images and resized the old images accordingly.

i noticed we were requiring images for cards that are no longer used, some of which were 2MB+. since this component loads on the main teacher dashboard, i went through and removed unused images (which are now obsolete anyway unless we resize them), and optimized the file size of the ones we're still using.

i also cleaned up our layout styling for LTR while i was at it.

Screen Shot 2021-04-30 at 10 56 12 AM
Screen Shot 2021-04-30 at 10 57 01 AM

## Links

- [spec](https://docs.google.com/document/d/1QrDuSVKQ1nPFujq_MAwSRBC2XXE1Of9a_OTGhNnNMZA/edit?ts=608af04f)
- jira ticket: [LP-1868]

[LP-1868]: https://codedotorg.atlassian.net/browse/LP-1868

### CODE REVIEW

File: apps/src/templates/studioHomepages/ImageResourceCard.jsx [hunk 1]
```diff
@@ -6,33 +6,44 @@ import color from '../../util/color';
 
 const styles = {
   card: {
-    overflow: 'hidden',
-    position: 'relative',
+    display: 'flex',
```
Comment (maureensturgeon) [inline apps/src/templates/studioHomepages/ImageResourceCard.jsx:9]: 👏

File: apps/src/templates/studioHomepages/ImageResourceCard.jsx [hunk 2]
```diff
@@ -41,57 +52,44 @@ const styles = {
...
-        <div style={{...styles.textbox, ...localeStyle, ...uncoverImage}}>
-          <div style={styles.title}>{title}</div>
-          <div style={styles.description}>{description}</div>
-          <br />
+      <div style={{...styles.card, ...(isRtl && styles.rtl)}}>
+        <div style={styles.textbox}>
+          <div>
+            <div style={styles.titleContainer}>
+              <div style={styles.title}>{title}</div>
+              <div style={styles.callout}>
```
Comment (maureensturgeon) [inline apps/src/templates/studioHomepages/ImageResourceCard.jsx]: Do we want to conditionally render this block if there is a callout?

File: apps/src/templates/studioHomepages/ResourceCard.jsx [hunk 3]
```diff
@@ -26,17 +26,22 @@ const styles = {
...
-    paddingBottom: 15,
     fontSize: 27,
-    width: '100%',
-    display: 'inline',
+    lineHeight: '29px',
     boxSizing: 'border-box'
   },
   titleSmall: {
-    width: '100%',
+    // width: '100%',
```
Comment (cforkish) [inline apps/src/templates/studioHomepages/ResourceCard.jsx]: just noticed this. i'm going to wait for the build to finish before pushing the fix.

Comment (maureensturgeon) [inline apps/src/templates/studioHomepages/ResourceCard.jsx]: Did you mean to leave this?

File: apps/src/templates/studioHomepages/ResourceCard.jsx [hunk 4]
```diff
@@ -131,7 +145,12 @@ class ResourceCard extends Component {
 
     return (
       <div style={cardStyles}>
-        <div style={titleStyles}>{title}</div>
+        <div style={styles.titleContainer}>
+          <div style={titleStyles}>{title}</div>
+          <div style={styles.callout}>
```
Comment (maureensturgeon) [inline apps/src/templates/studioHomepages/ResourceCard.jsx]: Do you still want to render this block if there's no callout?

Comment (cforkish) [reply apps/src/templates/studioHomepages/ResourceCard.jsx]: yeah i guess it's clearer if i make that conditional. i'll update when i remove that stray commented line after the build completes

**Review by maureensturgeon — APPROVED** [review]
Are we going to want to use those image files again in the near future? Are we sure we want to delete them?

**Comment by cforkish** [comment]
the images i deleted would need to be updated if we want to reuse them. i was thinking it wouldn't be too hard to find them again through git history, and i'm not a big fan of leaving unused resources in our codebase (especially not ~15MB's worth), but if you think it would make future devs' live's easier i can update and optimize them and put them back in in a followup PR.

**Comment by cforkish** [comment]
also worth noting -- there's a lot of duplicated code between `ResourceCard`, `ImageResourceCard`, and `VerticalImageResourceCard` that would be good to cleanup and consolidate, and it sure would be nice if non-engineers could update this content. so rather than addressing card reusability now, i'm going to add a jira task to revisit this space in a more holistic way.

**Comment by cforkish** [comment]
followup work: https://codedotorg.atlassian.net/browse/LP-1878

**Pull request merged**